test_runner/batch_others/test_pageserver_api.py

```diff
@@ -73,18 +76,39 @@ def test_pageserver_http_get_wal_receiver_success(zenith_simple_env: ZenithEnv):
     tenant_id, timeline_id = env.zenith_cli.create_tenant()
     pg = env.postgres.create_start(DEFAULT_BRANCH_NAME, tenant_id=tenant_id)
 
-    res = client.wal_receiver_get(tenant_id, timeline_id)
-    assert list(res.keys()) == [
-        "thread_id",
-        "wal_producer_connstr",
-        "last_received_msg_lsn",
-        "last_received_msg_ts",
-    ]
-
-    # make a DB modification then expect getting a new WAL receiver's data
+    def try_get_updated_msg_lsn(prev_msg_lsn: Optional[int]) -> int:
+        """Try to get the updated message's LSN from the WAL receiver.
+
+        This function raises an exception if the latest message's LSN is not found
+        or it hasn't been updated compared to the previous message's LSN.
+        """
+        res = client.wal_receiver_get(tenant_id, timeline_id)
```

Comment:
Nit: this docstring may become excessive after the function is reduced to a few lines as per comments below.